### DISCUSSION TITLE
RP2040 1.28 WAVESHARE TOUCH LCD

### DISCUSSION THREAD

**Discussion by tonluc123** [moononournation/Arduino_GFX#450] [Q&A]
Hello,
I'm using this library successfully for almost a year now. I used GC9A01 & GC9A01 with touch and it is great. Lately I purchased a RP2040 1.28 WAVESHARE TOUCH LCD which I thought it would be easy to manage as it uses the same hardware , but ... no luck.
I tried the clock example and used the Dev-Device-Declaration#waveshare-rp2040-lcd-128 but nothing comes on the screen.
Does anybody had any experience with this module? Any help would be appreciated.
Lucas

**Comment by tonluc123** [comment]
To be more accurate , this is the code
`
```cpp
#include <Arduino_GFX_Library.h>
#define GFX_BL 25
Arduino_DataBus *bus = new Arduino_RPiPicoSPI(8 /* DC */, 9 /* CS */, 10 /* SCK */, 11 /* MOSI */, 12 /* MISO */, spi1 /* spi */);
Arduino_GFX *gfx = new Arduino_GC9A01(bus, 12 /* RST */, 0 /* rotation */, true /* IPS */);

void setup() {
 Serial.begin(115200);
  gfx->begin();
  gfx->fillScreen(BLUE);
}

void loop() {
  // put your main code here, to run repeatedly:

}

```
`

**Reply by moononournation** [reply]
wild guess just not yet turn on backlight LED

**Reply by tonluc123** [reply]
Hello,
Thanks for your response. I included BL, but no change, see code bellow (maybe I did not implement it correct)
`#include <Arduino_GFX_Library.h>
#define GFX_BL 25
Arduino_DataBus *bus = new Arduino_RPiPicoSPI(8 /* DC */, 9 /* CS */, 10 /* SCK */, 11 /* MOSI */, 12 /* MISO */, spi1 /* spi */);
Arduino_GFX *gfx = new Arduino_GC9A01(bus, 12 /* RST */, 0 /* rotation */, true /* IPS */);

void setup() {
  pinMode(GFX_BL, OUTPUT);
  digitalWrite(GFX_BL, HIGH);
 Serial.begin(115200);
  gfx->begin();
  gfx->fillScreen(BLUE);
}

void loop() {
  // put your main code here, to run repeatedly:

}`
Lucas

**Reply by tonluc123** [reply]
I can see visually that bl is on
